hw/ip/otbn/dv/uvm/env/otbn_test_helpers.cc

```diff
@@ -0,0 +1,114 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+// Helper code for the UVM testbench. The corresponding SV declarations are in
+// otbn_env_pkg.sv.
+
+#include <cassert>
+#include <cstring>
+#include <dirent.h>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <sys/types.h>
+
+namespace {
+struct DirDeleter {
+  void operator()(DIR *dir) const {
+    // Ignore the return value from closedir: the only way it can fail is if
+    // dir was an invalid descriptor, which it isn't.
+    (void)closedir(dir);
+  }
+};
+
+typedef std::unique_ptr<DIR, DirDeleter> dir_ptr;
+
+class OtbnTestHelper {
+ public:
+  std::string dir_path_;
+  std::string file_path_;
+
+  OtbnTestHelper(const std::string dir_path) : dir_path_(dir_path) {}
+
+  int CountFilesInDir() const {
+    dir_ptr dir(opendir(dir_path_.c_str()));
+    if (!dir)
+      return 0;
+
+    int count = 0;
+    for (;;) {
+      struct dirent *entry = readdir(dir.get());
+      if (!entry)
+        break;
+
+      if (entry->d_type == DT_REG)
+        ++count;
+    }
```

Comment:
Is it worth checking the file extension pehaps, to make it more robust, in case you supply it a directory containing arbitrary sources?

Reply:
I wasn't sure whether to do that (since ELF files don't always have the same suffix: this ain't Windows! :-)) and I was also worried about silently filtering things ("whaddya mean, no files?! They're right here!").

Maybe this would be helpful, and it would make it easier if we want to have a "my_test_list.txt" in the same directory. I'll change the code to do the filtering, but make it print a warning to stderr whenever we see an unexpected file. Hopefully that will strike the right balance between helpful and mysterious :-D